### ISSUE TITLE
Allow custom quote character/no quote character when exporting to CSV

### ISSUE THREAD

**Issue by egmanoj** [jOOQ/jOOQ#3330]
Something like this:

```java
String csv = create.selectFrom(BOOK).fetch().formatCSV('\t', "{null}", NO_QUOTE_CHARACTER);
```

**Comment by lukaseder** [comment]
Hmm, I've had an attempt at looking what needs to be changed. What does it mean according to you to not have a quote character? I mean, how would you escape / quote values that contain `,`, or `\n`, or other special characters?

**Comment by lukaseder** [comment]
Requirements are still incomplete. Feel free to reopen if needed, with additional requirements about this feature.

**Comment by rikiya-yamamoto** [comment]
I wish support for quote by double quote.

ex.
"head1","head2","head3"
"data1","data2","data3"

**Comment by lukaseder** [comment]
@rikiya-yamamoto: Thanks for your feedback. I see, so the criticism here is about enforcing the application of the quote character, even when it is not strictly needed?

**Comment by rikiya-yamamoto** [comment]
I need this function.

**Comment by lukaseder** [comment]
OK :)

**Comment by lukaseder** [comment]
This is available via the new `CSVFormat` API (see #4818). There is now a possibility to specify whether quoting should be done at all, and if yes, with what character.
